File: c/simpson.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

float f(float x){
  return sqrt(1-x*x);
}

float simps(float a, float b, float n) {
  float h,sum1=0,sum2=0,sum,y0,yn;
  int i;
  h = (b-a)/n;
  y0=f(a+0*h);
  yn=f(a+n*h);
  for(i=1; i<n; i++) {
    if(i%2 == 0)
      sum1 = sum1 + f(a+i*h);
    else
      sum2 = sum2 + f(a+i*h);
  } 
  sum = (h/3)*(y0 + yn + 2*sum1 +4*sum2);
  return sum;
}

int main(){
  float Pi;
  Pi = 4*simps(0, 1, 1000);
  printf("Valor:     %f", Pi);
  printf("\nPrecisão: %f\n",(Pi-M_PI));
	return 0;
}

File: fichas_de_estudo/rust/simpson_simples/simpson.rs

fn f(x: f32) -> f32{
    let aux = 1.0 - (x*x);
    return aux.sqrt();
}

fn simps(a: f32, b: f32, n: i32) -> f32{
    let h;
    let y0;
    let yn;
    let sum;
    let mut sum1 = 0.0;
    let mut sum2 = 0.0;
    h = (b-a)/(n as f32);
    y0 = f(a+(0.0*h));
    yn = f(a+((n as f32)*h));

    for i in 1..n {
        if i%2 == 0 {
            sum1 = sum1 + f(a+((i as f32)*h));
        } else {
            sum2 = sum2 + f(a+((i as f32)*h));
        }
    }

    sum = (h/3.0)*(y0 + yn + (2.0*sum1) + (4.0*sum2));
    return sum;
}

fn main(){
    let mat_pi = std::f32::consts::PI;
    let pi;
    pi = 4.0*simps(0.0, 1.0, 1000);
    println!("Valor:     {}", pi);
    println!("Precisão: {}", (pi-mat_pi));
}